### DISCUSSION TITLE
Clear Button in TextBox, ComboBox and DataGridComboBoxColumn, DataGridTextBoxColumn

### DISCUSSION THREAD

**Discussion by hxhgxy** [microsoft/WinUI-Gallery#1148] [Q&A]
Could you please make somes examples for clear button in TextBox, ComboBox and DataGridComboBoxColumn, DataGridTextBoxColumn?

I cannot do it for DataGridComboBoxColumn in WPF, while others are ok.

thanks a lot!

**Comment by marcelwgn** [comment]
It seems that you are requesting samples for WPF controls, is that correct? In that case, I'm afraid this is the wrong repository, this repository is for the WinUI Gallery which covers WinUI 3.

**Comment by hxhgxy** [comment]
I'm not asking you for WPF controls. I noticed textbox in WINUI 3 has clear button. can combobox and DataGridComboBoxColumn have the same figure?

i cannot do it in WPF, so that to see if can do it in WINUI 3.

**Comment by marcelwgn** [comment]
I'm afraid the only solution to get a clear button on ComboBox/DataGridComboBoxColumn would be to retemplate ComboBox or override the `ComboBoxTextboxStyle` resource. You can find the current WinUI 2 implementation of the ComboBoxTextBoxStyle [here](https://github.com/microsoft/microsoft-ui-xaml/blob/84bc7ac6672a879ed9e7d879bb142b716d97507b/dev/ComboBox/ComboBox_themeresources.xaml#L967) and the current TextBox style of WinUI 2 [here](https://github.com/microsoft/microsoft-ui-xaml/blob/84bc7ac6672a879ed9e7d879bb142b716d97507b/dev/CommonStyles/TextBox_themeresources.xaml#L207).

So there is a way to do this, but it would require retemplating which will be a bit of work.

**Reply by Jay-o-Way** [reply]
@chingucoding I think the subject is to insert a <kbd>✖️</kbd> button. Not the style/color of it. 😉

**Reply by marcelwgn** [reply]
To properly insert it inside the control, I think you need to retemplate it, otherwise it wouldn't be part of the control
